platform/migration/platform-migration/src/test/java/org/codice/ddf/configuration/admin/ConfigurationFilesPollerTest.java

```diff
@@ -86,234 +77,40 @@ public void setup() {
-                mock(Path.class));
-        ConfigurationFilesPoller configurationFilesPoller = createConfigurationFilesPoller(
-                mockWatchService,
-                mockConfigurationDirectory);
-        configurationFilesPoller.register(mockChangeListener);
-
-        // Perform Test
-        configurationFilesPoller.run();
-
-        // Verify
-        verify(mockChangeListener, never()).notify(any(Path.class));
-    }
-
-    /**
-     * Verify that file deletion events are not sent to the ChangeListener.
-     */
-    @Test
-    public void testRunFileDeletedChangeListenerNotNotified() throws Exception {
-        // Setup
-        Path mockPath = getMockPath(PID, FILE_EXT);
-        WatchEvent<?> mockWatchEvent = getMockWatchEvent(mockPath,
-                StandardWatchEventKinds.ENTRY_DELETE);
-        List<WatchEvent<?>> watchEvents = getSingleMockWatchEvent(mockWatchEvent);
-        WatchKey mockWatchKey = getMockWatchKey(watchEvents);
-        WatchService mockWatchService = getMockWatchService(mockWatchKey);
-        Path mockConfigurationDirectory = getMockConfigurationDirectoryPath(mockPath,
-                mock(Path.class));
-
-        ConfigurationFilesPoller configurationFilesPoller = createConfigurationFilesPoller(
-                mockWatchService,
-                mockConfigurationDirectory);
-        configurationFilesPoller.register(mockChangeListener);
-
-        // Perform Test
-        configurationFilesPoller.run();
-
-        // Verify
-        verify(mockChangeListener, never()).notify(any(Path.class));
-    }
-
-    /**
      * Verify that the ChangeListener is notified on file creation events.
      */
     @Test
     public void testRunFileCreatedChangeListenerNotified() throws Exception {
         // Setup
-        Path mockPath = getMockPath(PID, FILE_EXT);
-        WatchEvent<?> mockWatchEvent = getMockWatchEvent(mockPath,
-                StandardWatchEventKinds.ENTRY_CREATE);
-        List<WatchEvent<?>> watchEvents = getSingleMockWatchEvent(mockWatchEvent);
-        WatchKey mockWatchKey = getMockWatchKey(watchEvents);
-        WatchService mockWatchService = getMockWatchService(mockWatchKey);
-        Path resolvedPath = mock(Path.class);
-        Path mockConfigurationDirectory = getMockConfigurationDirectoryPath(mockPath, resolvedPath);
-        ConfigurationFilesPoller configurationFilesPoller = createConfigurationFilesPoller(
-                mockWatchService,
-                mockConfigurationDirectory);
-        configurationFilesPoller.register(mockChangeListener);
+        Path mockPath = Files.createTempDirectory("foo");
+        ConfigurationFilesPoller configurationFilesPoller = new ConfigurationFilesPoller(mockPath,
+                FILE_EXT);
 
         // Perform Test
-        configurationFilesPoller.run();
+        configurationFilesPoller.register(mockChangeListener);
+        Files.createTempFile(mockPath, "foo", FILE_EXT);
+        Thread.sleep(5000);
```

Comment:
❗️Instead of testing with real files and having to deal with sleeps and such, couldn't we simply test the individual notification methods and make sure they do what they're supposed to? That may require injecting and mocking out the `FileAlterationMonitor` object, but that sounds like a better option imho.